nova/db/sqlalchemy/api.py

```diff
@@ -411,9 +411,26 @@ def convert_objects_related_datetimes(values, *datetime_keys):
 
 
 def _sync_instances(context, project_id, user_id):
-    return dict(zip(('instances', 'cores', 'ram'),
-                    _instance_data_get_for_user(context, project_id, user_id)))
-
+    query = '''
+      SELECT instance_type_id, COUNT(id), SUM(vcpus), SUM(memory_mb),
```

Comment:
`instance_type_id` is an internal id, which is `InstanceType#id`, which is not the same as the public `InstanceType#flavorid`.
To be consistent it, would have to be the latter.

Could you do a join over instance_types and generate `quota_key_instances` as you do in  nova/compute/api.py:943, and summarise in the db?
I think, that would probably avoid a couple of roundtrips to the db.